### DISCUSSION TITLE
How to send pdf file to browser for download. Currently my source code is in Local .  How the pdf will be sent to browser when django application deployed to render webs service.

### DISCUSSION THREAD

**Discussion by rohinitechsolutions** [py-pdf/fpdf2#1255] [General]
Using this version of fpdf2: fpdf2==2.7.4 
Have tried all these options
            # pdf.output(my_file_path)
            # bytes(pdf.output(), encode='latin-1')
            # pdf.output(my_file_path, "I")
            # pdf.output('abc.pdf','F')
            pdf.output('D','filename.pdf')
            # pdf.output('D', custname + '-'+ invoicenumber +'.pdf')
            # pdf.output('F', custname + '-'+ invoicenumber +'.pdf')
            # pdf.output(custname + '-'+ invoicenumber +'.pdf')
            
I want to send the pdf file to browser and it should work in development as well as in production server(deploying in render web service) .
Any help will be greatly appreciated

**Comment by andersonhc** [comment]
Hello @rohinitechsolutions 

Did you check [the documentation regarding Django?](https://py-pdf.github.io/fpdf2/UsageInWebAPI.html)

Please let us know if the example there is clear and answers your question.

**Reply by rohinitechsolutions** [reply]
I have used the code but still the PDF is not sent to browser for download . 
<img width="642" alt="image" src="https://github.com/user-attachments/assets/1a48445e-3411-4324-aec7-28b8b852a4aa">

**Reply by andersonhc** [reply]
Are you calling `report2(request)`?
I'm asking because the faded color on your IDE suggests it's a function you're not calling.
